common/src/main/java/com/sequenceiq/cloudbreak/util/PasswordUtil.java

```diff
@@ -1,28 +1,31 @@
 package com.sequenceiq.cloudbreak.util;
 
-import java.math.BigInteger;
-import java.security.MessageDigest;
-import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
 
 import org.apache.commons.lang3.RandomStringUtils;
 
 public class PasswordUtil {
 
-    private static final int PWD_LENGTH = 128;
+    private static final int PWD_LENGTH = 26;
 
-    private static final int RADIX = 32;
+    private static final SecureRandom SECURE_RANDOM = new SecureRandom();
 
     private PasswordUtil() {
     }
 
     public static String generatePassword() {
-        try {
-            MessageDigest messageDigest = MessageDigest.getInstance("MD5");
-            String raw = RandomStringUtils.randomAscii(PWD_LENGTH);
-            byte[] digest = messageDigest.digest(raw.getBytes());
-            return new BigInteger(1, digest).toString(RADIX);
-        } catch (NoSuchAlgorithmException e) {
-            throw new RuntimeException(e);
-        }
+        return getRandomLettersAndNumbersWithCount(PWD_LENGTH, SECURE_RANDOM);
+    }
+
+    public static String getRandomLettersAndNumbersWithCount(int count, SecureRandom secureRandom) {
+        return RandomStringUtils.random(count, 0, 0, true, true, null, secureRandom);
+    }
+
+    public static String getRandomAlphabetic(int count) {
+        return RandomStringUtils.random(count, 0, 0, true, false, null, SECURE_RANDOM);
+    }
+
+    public static String getRandomNumeric(int count) {
```

Comment:
It's hard to grasp all these parameters `RandomStringUtils.random(count, 0, 0, true, false, null, SECURE_RANDOM)`;

As I see the parameters that we want to configure are `count, letters, numbers, random`.

Can you please create a separate method and use that in upper methods:

```java 
private static String generate(int count, boolean letters, boolean numbers) {
    return RandomStringUtils.random(count, 0, 0, letters, numbers, SECURE_RANDOM);
}
```